### PR TITLE
Migrate to using new version of pgx and associated libs

### PR DESCRIPTION
**Pull request by stlava**

### CODE REVIEW

File: replication/client/client.go [hunk 1]
```diff
@@ -45,7 +47,7 @@ var (
 
 func init() {
 	logger.SetOutput(os.Stdout)
-	logger.SetLevel(logrus.InfoLevel)
+	logger.SetLevel(logrus.DebugLevel)
```
Comment (diranged) [inline replication/client/client.go:50]: On purpose?

File: replication/client/client.go [hunk 2]
```diff
@@ -235,19 +235,31 @@ func (c *Replicator) Start(progressChan <-chan uint64) {
...
+		message, rplErr = pgConn.ReceiveMessage(replicationCtx)
 	}()
 
-	if message == nil || message.ServerHeartbeat == nil {
+
+	cd, ok := message.(*pgproto3.CopyData)
+	if !ok {
+		log.Errorf("Received unexpected message: %#v", message)
+		return
+	}
```
Comment (diranged) [inline replication/client/client.go:246]: So what happens if this returns?

File: replication/client/client.go [hunk 3]
```diff
@@ -280,51 +292,63 @@ func (c *Replicator) Start(progressChan <-chan uint64) {
...
+		// Begin reading message
 		if message == nil {
 			log.Debug("message was nil")
 			continue
 		}
 
+		cd, ok := message.(*pgproto3.CopyData)
+		if !ok {
+			log.Errorf("Received unexpected message: %#v", message)
+			continue
```
Comment (diranged) [inline replication/client/client.go:340]: What happens here?

File: replication/client/client.go (hunk 1 above)
Comment (stlava) [reply replication/client/client.go:50]: Yeah just for now while I debug things. Will remove before merging.

File: replication/client/client.go (hunk 2 above)
Comment (stlava) [reply replication/client/client.go:246]: Bifrost shuts down. This is a new error case that didn't exist in the old library. I haven't yet added test cases for these yet.

File: replication/client/client.go (hunk 3 above)
Comment (stlava) [reply replication/client/client.go:340]: Same as above. I haven't handled this yet but it will be part of this PR.

**Pull request merged**